File: libraries/lib_retrofit/src/main/java/com/example/william/my/core/retrofit/base/ResponseCallback.java
package com.example.william.my.core.retrofit.base;

import com.example.william.my.core.retrofit.exception.ApiException;

public interface ResponseCallback<T> {

    /**
     * onLoading
     */
    void onLoading();

    /**
     * onResponse
     *
     * @param response
     */
    void onResponse(T response);

    /**
     * onFailure
     *
     * @param e
     */
    void onFailure(ApiException e);
}

File: libraries/lib_retrofit/src/main/java/com/example/william/my/core/retrofit/callback/ResponseCallback.kt
package com.example.william.my.core.retrofit.callback

import com.example.william.my.core.retrofit.exception.ApiException

interface ResponseCallback<T> {
    /**
     * onLoading
     */
    fun onLoading()

    /**
     * onResponse
     *
     * @param response
     */
    fun onResponse(response: T)

    /**
     * onFailure
     *
     * @param e
     */
    fun onFailure(e: ApiException)
}